components/AppLink.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <component :is="linkTag" v-bind="attr">
+    <slot />
+    <v-icon
+      v-if="_showIcon"
+      class="ExternalLinkIcon"
+      :class="iconClass"
+      :size="_iconSize"
+      :aria-label="this.$t('別タブで開く')"
+      role="img"
+      :aria-hidden="false"
+    >
+      {{ iconType }}
+    </v-icon>
+  </component>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { isExternal } from '@/utils/urls.ts'
+
+type ExternalAttr = {
+  to?: String
+  href?: String
+  target?: String
+  rel?: String
+  class: String
+}
+
+export default Vue.extend({
+  props: {
+    to: {
+      type: String,
+      required: true,
+    },
+    showIcon: {
+      type: Boolean,
+      default: null,
+    },
+    iconSize: {
+      type: Number,
+      default: 12,
+    },
+    iconType: {
+      type: String,
+      default: 'mdi-open-in-new',
+    },
+    iconClass: {
+      type: String,
+      default: '',
+    },
+  },
+  computed: {
+    linkTag(): string {
+      return isExternal(this.to) ? 'a' : 'nuxt-link'
+    },
+    attr(): ExternalAttr {
+      if (isExternal(this.to)) {
+        return {
+          href: this.to,
+          target: '_blank',
+          rel: 'noopener noreferrer',
+          class: 'ExternalLink',
+        }
+      } else {
+        return {
+          to: this.to,
+          class: 'Link',
+        }
+      }
+    },
+    _showIcon(): boolean {
+      // 指定がない場合、外部なら表示、内部なら表示しない
+      if (this.showIcon == null) {
+        return isExternal(this.to)
+      } else {
+        return this.showIcon
+      }
```

Comment:
[Null合体演算子](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)が使用できそうです（動作未確認）。

```suggestion
      return this.showIcon ?? isExternal(this.to)
```

Reply:
これも知りませんでした。動作確認含めてやってみます。

Reply:
対応しました。算出プロパティ化したので、以下になりました。スッキリです。
```
return this.showIcon ?? this.isExternal
```

Reply:
こちらはIE11でも大丈夫そうですかね...？

Reply:
ど、どうでしょう。IE11が手元にないので確認できていません。

コンパイルされて以下のようになっているようです。

```
eval("

中略

function _showIcon() {
      var _a; // 指定がない場合、外部なら表示、内部なら表示しない


      return (_a = this.showIcon) !== null && _a !== void 0 ? _a : this.isExternal;
    },

中略

");
```
(\nを改行に変換してます。)

内部的には三項演算子になっているので、問題ないと思われます。
https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Operators/Conditional_Operator#Browser_compatibility

Reply:
おお、大丈夫そうですね！プレビューで確認しておきますね。